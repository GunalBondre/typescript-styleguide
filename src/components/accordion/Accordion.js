import React from 'react';
import PropTypes from 'prop-types';
import { AccordionWrapper } from './Accordion.style';
import AccordionItem from './children/AccordionItem';
import AccordionHeading from './children/AccordionHeading';
import AccordionDescription from './children/AccordionDescription';
import { useState } from 'react';

const Accordion = ({ children, alwaysOpen, multiple, ...restProps }) => {
	const [active, setActive] = useState(null);

	return (
		<div>
			<AccordionWrapper
				alwaysOpen={alwaysOpen}
				children={children}
				multiple={multiple}
				{...restProps}
			>
				{React.Children.map(children, (child, index) => {
					return React.cloneElement(child, {
						index,
						active,
						setActive,
						multiple,
					});
				})}
			</AccordionWrapper>
		</div>
	);
};

Accordion.Item = AccordionItem;
Accordion.Heading = AccordionHeading;
Accordion.Description = AccordionDescription;

Accordion.propTypes = {
	/**
	 * Children denotes accordion childrens
	 */
	children: PropTypes.node,
	/**
	 * always opem denotes accordion open or not
	 */
	alwaysOpen: PropTypes.bool,
};

export default Accordion;
