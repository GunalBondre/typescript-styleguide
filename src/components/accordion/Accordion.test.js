import { render, screen } from '@testing-library/react';
import Accordion from './Accordion';

describe('Accordion component:', () => {
	it('should render the accordion component', () => {
		const { container } = render(
			<Accordion>
				<Accordion.Item key={1}>
					<Accordion.Heading>Accordion 1</Accordion.Heading>
					<Accordion.Description>
						Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
					</Accordion.Description>
				</Accordion.Item>
			</Accordion>
		);

		expect(container).toMatchSnapshot();
		expect(screen.getByTestId('accordion')).toBeInTheDocument();
	});
});
