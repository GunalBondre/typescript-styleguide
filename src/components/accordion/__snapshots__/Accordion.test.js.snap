// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component: should render the accordion component 1`] = `
<div>
  <div>
    <div
      class="sc-bczRLJ hKfuKe"
      data-testid="accordion"
    >
      <div
        class="sc-gsnTZi fpCuiP"
      >
        <div
          class="sc-hKMtZM kJjoWu"
        >
          <div
            class="sc-eCYdqJ"
          >
            Accordion 1
          </div>
          <div
            class="sc-jSMfEi hsBVmt"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-dkzDqf iTkAXJ"
        />
      </div>
    </div>
  </div>
</div>
`;
